src/api/matrix.js

```diff
@@ -0,0 +1,36 @@
+import Http from '../http';
+import Navigator from '../navigator';
+
+class Matrix {
+  static async getMatrixDistances(routingOptions={}) {
+    if (!routingOptions.origins) {
+      const { latitude, longitude } = await Navigator.getCurrentPosition();
+      routingOptions.origins = { latitude, longitude };
+    }
+
+    let {
+      origins,
+      destinations,
+      mode,
+      units,
+    } = routingOptions;
+
+    origins = origins;
+    mode = mode;
```

Comment:
don't think these two lines are necessary, you're already destructuring these vars from `routingOptions`